### PR TITLE
Only build stack for http errors when the feature is enabled

### PR DESCRIPTION
**Pull request by waltjones**
This PR ensures stacks for reporting xhr and fetch errors are built only when that feature is enabled.

Calling `stack()` on an Error object causes it to show up at `Error.prepareStackTrace`. That's unavoidable, but limiting it as much as possible helps the anonymous errors feature (which relies on `Error.prepareStackTrace`) work more reliably in more situations.

### CODE REVIEW

File: src/browser/telemetry.js [hunk 1]
```diff
@@ -354,7 +354,9 @@ Instrumenter.prototype.instrumentNetwork = function() {
           }
         }
         self.captureNetwork(metadata, 'fetch', undefined);
-        metadata.stack = (new Error()).stack;
+        if (self.trackHttpErrors()) {
+          metadata.stack = (new Error()).stack;
```
Comment (jondeandres) [inline src/browser/telemetry.js:358]: Should we send `stack = null` here or is ok not sending the `stack` property at all?

Comment (waltjones) [reply src/browser/telemetry.js:358]: I think not setting at all would be the preference.

Comment (jondeandres) [reply src/browser/telemetry.js:358]: :+1:

**Comment by waltjones** [comment]
@jondeandres Thank you!

**Pull request merged**